frog/responsive-images.rkt

```diff
@@ -36,32 +39,68 @@
 
 (module+ test
   (when magick-available?
-    (parameterize ([top example])
+    (parameterize ([top example]
+                   [current-verbosity 99])
       (check-eq? (image-width (build-path (www/img-path) "800px-image.gif")) 800))))
 
-(define/contract (resize-image in new-width out-path)
-  (path? number? path? . -> . boolean?)
-  (prn1 "Shrinking ~a to ~a pixels... " (abs->rel/www in) new-width)
+(define/contract (resize-image input new-width out-path)
+  (path? number? path? . -> . void?)
+  (prn1 "Shrinking ~a to ~a pixels... " (abs->rel/www input) new-width)
   ;; Imagemagick options from
   ;; https://www.smashingmagazine.com/2015/06/efficient-image-resizing-with-imagemagick/
-  (apply system* mogrify
+  (define args
          `("-filter" "Triangle" "-define" "filter:support=2"
            "-unsharp" "0.25x0.08+8.3+0.045" "-dither" "None" "-posterize" "136"
            "-quality" "82" "-define" "jpeg:fancy-upsampling=off"
            "-define" "png:compression-filter=5" "-define" "png:compression-level=9"
            "-define" "png:compression-strategy=1" "-define" "png:exclude-chunk=all"
            "-interlace" "none" "-colorspace" "sRGB"
            "-thumbnail" ,(number->string new-width)
-           "-path" ,out-path ,in)))
+           "-path" ,out-path ,input))
+  ;; Make simple job server using dispatcher thread and thread mailboxes?
+  ;;
+  ;; One problem with the async approach is that if Frog is killed before
+  ;; subprocesses are finished they will not be triggered again if Frog is
+  ;; invoked again and the source post has not been touched. Ideally we would
+  ;; trap SIGINT and write out unfinished work to disk, or at least
+  ;; detect that work was finished prematurely and restart everything somehow.
+  (let-values ([(proc in out err) (apply subprocess
+                                         (current-output-port)
+                                         (current-input-port)
```

Comment:
You you may notice that I never explicitly close these ports, instead relying on the OS the close them once subprocess is finished. I think this should be OK.